fire-cakes/pancakes
# Support game creation and inital joining

### What does this PR do?
This PR represents a major overhaul of the schema. It also closes up some loose ends related to creating a game and the first player joining it.

There is now a streamlined option to create an account and make a game from the UI. No more need for running Game.create from the command line.

This PR also introduces the `available` scope on the game model uses it in the Games#index action on the controller to assign the variable `@avaliable_games`, which is used on the index page.

As of this PR the creator of this game is automatically assigned to be the game's `white_player`. The relationship between games and players was also changed to flow more simply.

On the game model a game has two players, one of each color (the class_name and foreign_key help rails know where to place each):
```
has_one :white_player, class_name: 'Player', foreign_key: 'white_player_id'
has_one :black_player, class_name: 'Player', foreign_key: 'black_player_id'
```

On the pieces model, a player belongs to a game:

```
belongs_to :game
```

Please leave a comment with any questions! Tested locally and seems to be working as expected


### How do I manually test this?
1. After checking out you will need to rewrite your database.
I changed old migrations rather than writing a slew of new ones to keep the code easier to follow and tell a more logical story. This means that you will need to run the following series of commands from the command line to reconstitute your DB.:
- `rake db:drop`
- `rake db:create`
- `rake db:migrate`

2. Go to the index page
3. Create an account
4. Start a new game
5. See the board you just made!

### GIF for how this PR makes me feel
![](https://media.giphy.com/media/AXZfaR6wACdIk/giphy.gif)
